common/src/serialize.rs

```diff
@@ -25,6 +25,63 @@ impl io::Write for Counter {
     }
 }
 
+/// The configuration for binary serialization.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct BinarySerializableConfig {
+    /// The precision with which DateTime is de/serialized
+    pub date_time_precision: DateTimePrecision,
+}
+
+/// Trait for configurable binary serialization
+pub trait ConfigurableBinarySerializable: fmt::Debug + Sized {
```

Comment:
The introduction of  ConfigurableBinarySerializable seems quite overkill.

I understand this is an attempt to be future proof, but let's wait for the need to resurface 2 or 3 times before going there.

Also, I don't think common is the right place for this trait. We could have it closer to the BinaryDocumentSerializer

Reply:
Yes, it's kinda overkill currently, I removed it